src/generator.rs

```diff
@@ -45,6 +45,35 @@ impl Generator {
         self.generate_from_datetime(crate::time_utils::now())
     }
 
+    /// Generate a new Ulid. Each call is guaranteed to provide a Ulid with a larger value than the
+    /// last call. The time part of the returned [`Ulid`] can be up to one millisecond in the future,
+    /// under reasonable assumptions about processor speeds.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the previously returned value was already the highest possible ulid.
+    /// This should not happen before year 91000 of the gregorian calendar at least.
+    ///
+    /// ```rust
+    /// use ulid::Generator;
+    /// let mut gen = Generator::new();
+    ///
+    /// let ulid1 = gen.generate_overflowing();
+    /// let ulid2 = gen.generate_overflowing();
+    ///
+    /// assert!(ulid1 < ulid2);
+    /// ```
+    pub fn generate_overflowing(&mut self) -> Ulid {
```

Comment:
If a single version of `generate_overflowing` is going to be added I'd prefer `generate_from_datetime_with_source_overflowing`. This would give the user the most choice.

However, looking at this code again, it may make sense to make the overflow value part of `MonotonicError`. If you're up for that change it should provide the best overall API here.

Reply:
Nevermind this doesn't work since it wouldn't store into `previous`.

Reply:
Considering the reason to be of `generate_overflowing` is for people who just want to say "I don't care", I think only having `generate_from_datetime_with_source_overflowing` would be counter-productive: after all, it's reasonably easy to replace it:
```rust
gen.generate_from_datetime_with_source_overflowing(time, source);
// is equivalent to
gen.generate_from_datetime_with_source(time, source)
    .unwrap_or_else(|| gen.generate_from_datetime_with_source(time + Duration::from_millis(1), source).unwrap());
```

So IMO this is the place where there's literally the least to gain, compared to `generate_overflowing` that does gain a lot compared to trying to reimplement it manually. Now, if you'd rather have all the variants, I'm all good with adding them; I was just trying to avoid combinatorial explosion of the API :)

I'll try to handle your other review comments soon! In the meantime please let me know your thoughts about the API you'd prefer so I can adjust this PR accordingly :)

Reply:
@dylanhart WDYT about this question?